### PR TITLE
Create `UserRoleViewModel`

### LINKED ISSUE #165: Create `UserRoleViewModel`
## DoD
- Change the name from `StartScreenViewModel` to `UserRoleViewModel`.
- Add the following methods to the `UserRoleManager`:
   - `func resetCurrentRole()`
   - `func setOwnerRole()`
   - `func setSitterRole()`

### PR DESCRIPTION
**Pull request by marinkman**
## Summary
This PR is a part of the work to resolve #165 

## Changes
- Change the class name from `StartScreenViewModel` to `UserRoleViewModel`
- Add the following methods to the `UserRoleViewModel`:
   - `func resetCurrentRole()`
   - `func setOwnerRole()`
   - `func setSitterRole()`
- Add tests to test these methods

## Checks to complete
- [x] I've built and run my changes, and no warnings or errors occurred.
- [x] All existing tests passed with my changes.
- [x] My code follows our style guide.
- [x] I've performed a self-review of my code.
- [x] PR's title has a conventional commit style (short and descriptive).
- [x] Every non-private interface is documented properly.
- [x] I've added tests for my code (if it makes sense).
- [x] PR has assignee, reviewers, milestone, it is properly linked to the project and to the issue.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -23,6 +23,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/).
 - Add the `SitterInfoCardView` - [#112](https://github.com/ios-course/ironfoudation-team-project/pull/158)
 - Create `PurpleCapsuleOfInfinityWidth` as `ButtonStyle`. Create `LoginView` and apply in the `WoofApp` to display when running the app when the user role is not selected - [#160](https://github.com/ios-course/ironfoudation-team-project/pull/160)
 - Add the `EditSitterInformationView` and the reusable element `TextEditorWithPlaceholder` - [#159](https://github.com/ios-course/ironfoudation-team-project/pull/159)
+- Change the class name from `StartScreenViewModel` to `UserRoleManager`. Add methods to the `UserRoleManager` to set and reset user's role in the application - [#166](https://github.com/ios-course/ironfoudation-team-project/pull/166)
```
Comment (kirpichonok) [inline CHANGELOG.md:26]: Should we place it to section `Changed`?

Comment (marinkman) [reply CHANGELOG.md]: Yes, moved.

**Review by NadzeyaShpakouskaya — CHANGES_REQUESTED** [review]
Do we test the UserRoleManager properly and enough?
 if I delete/commented the didSet observer, your tests are still passing
```swift
final class UserRoleManager: ObservableObject {
    /// The current user role that defines the app flow.
    @Published var userRole: Role {
        didSet {
//            PreferencesHandler.set(userRole: userRole)
        }
    }
...
}
```

Do we need to add a method in the test target which will read data from storage to verify that our observer does his job as expected?

File: Woof/Woof/ViewModel/UserRoleManager.swift [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import Foundation
+
+/// Responsible for managing and actual state of the current user role in the application.
+final class UserRoleManager: ObservableObject {
```
Comment (iBYarchitect) [inline Woof/Woof/ViewModel/UserRoleManager.swift:4]: Is it worth calling a manager a class that is essentially a view model?

**Review by iBYarchitect — CHANGES_REQUESTED** [review]
In general, everything is fine, but there are a couple of questions:

File: Woof/Woof/ViewModel/UserRoleManager.swift (hunk 2 above)
Comment (marinkman) [reply Woof/Woof/ViewModel/UserRoleManager.swift]: Changed the class name to `UserRoleViewModel`

File: Woof/WoofTests/ViewModel/UserRoleManagerTests/UserRoleManagerResetCurrentRoleTests.swift [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    func testMethodResetsTheOwnerRole() {
+        // Given
+        userRoleManager.setOwnerRole()
+
+        // When
+        userRoleManager.resetCurrentRole()
+
+        // Then
+        XCTAssertEqual(userRoleManager.userRole, .none)
```
[the remark this answers is not in the document]
Comment (marinkman) [reply Woof/WoofTests/ViewModel/UserRoleManagerTests/UserRoleManagerResetCurrentRoleTests.swift]: Added.

**Review by iBYarchitect — APPROVED** [review]
LGTM

**Review by julyakmita — APPROVED** [review]
Good improvement! 
Looks good.
LGTM.

File: Woof/WoofTests/ViewModel/UserRoleViewModelTests/UserRoleViewModelTests.swift [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import XCTest
+
+final class UserRoleViewModelTests: XCTestCase {
+    func testUserRoleManagerExistInAPI() {
+        _ = UserRoleViewModel()
```
Comment (NadzeyaShpakouskaya) [inline Woof/WoofTests/ViewModel/UserRoleViewModelTests/UserRoleViewModelTests.swift:5]: Please update the name of the test methods, they don't match after renaming.

**Review by NadzeyaShpakouskaya — APPROVED** [review]
LGTM, just please fix the name of the test methods in `UserRoleViewModelTests`

File: Woof/WoofTests/ViewModel/UserRoleViewModelTests/UserRoleViewModelTests.swift (hunk 4 above)
Comment (marinkman) [reply Woof/WoofTests/ViewModel/UserRoleViewModelTests/UserRoleViewModelTests.swift]: Done.

**Pull request merged**